lib/routemap.c

```diff
@@ -1424,6 +1414,12 @@ enum rmap_compile_rets route_map_add_match(struct route_map_index *index,
 	} else
 		compile = NULL;
 
+	if (compile && cmd->func_get_rmap_rule_key)
```

Comment:
could you add a comment here describing why is this needed?

Reply:
done, also added the comment in bgp_routemap.c for route_match_get_community_key